test/unit/unit_socket.c

```diff
@@ -39,13 +39,13 @@ START_TEST (test_socket)
     s = pico_socket_open(PICO_PROTO_IPV4, 99, NULL);
     fail_if(s != NULL, "Error got socket wrong parameters");
 
-    s = pico_socket_open(PICO_PROTO_IPV4, -109, NULL);
+    s = pico_socket_open(PICO_PROTO_IPV4, (uint16_t)-109, NULL);
```

Comment:
Yes, I know it's dirty to shut the compiler up like this.
But if I understand correctly, the intend of the original test is to test these functions with negative numbers/invalid params. The compiler complained about this (because why would you want to pass a negative number to a function accepting only uint*_t?).
I'm not sure if this is a sane test, but I decided to cast the param to keep the test.
If you have another suggestion, let me know and I'll change where necessary.